atlasdb-config/src/main/java/com/palantir/atlasdb/http/NotCurrentLeaderExceptionMapper.java

```diff
@@ -20,43 +20,35 @@
 import javax.ws.rs.core.Response;
 import javax.ws.rs.ext.ExceptionMapper;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
+import com.palantir.conjure.java.api.errors.QosException;
 import com.palantir.leader.NotCurrentLeaderException;
-import com.palantir.logsafe.SafeArg;
-import com.palantir.logsafe.exceptions.SafeIllegalStateException;
 
 /**
- * Convert {@link NotCurrentLeaderException} into a 503 status response.
+ * Converts {@link NotCurrentLeaderException} into appropriate status responses depending on the user's
+ * {@link AtlasDbHttpProtocolVersion}. This is a 503 response in {@link AtlasDbHttpProtocolVersion#LEGACY_OR_UNKNOWN}.
  *
  * @author carrino
  */
 public class NotCurrentLeaderExceptionMapper implements ExceptionMapper<NotCurrentLeaderException> {
-    private static final Logger log = LoggerFactory.getLogger(NotCurrentLeaderExceptionMapper.class);
-
     @Context
     private HttpHeaders httpHeaders;
 
-    /**
-     * Returns a 503 response, with body corresponding to the serialized exception.
-     */
+    private static final HttpProtocolAwareExceptionTranslator<NotCurrentLeaderException> translator = new
```

Comment:
is this for tests?

Reply:
No, this is where the logic for deciding what the things actually map to lives

Reply:
talked about this in #4192, abstract class means we can just pass in the lambda handler most of this away, if anything we can still have the translator but as a packpage private static class (if you're testing it, can be private if not and just rely on integration tests)